pkg/thumbnail/thumbnail.go

```diff
@@ -84,3 +84,42 @@ func BuildStaticThumbnail(inputBuf []byte, resp *http.Response) ([]byte, error)
 
 	return outputBuf, nil
 }
+
+func BuildAnimatedThumbnail(inputBuf []byte, resp *http.Response) ([]byte, error) {
+	image, err := vips.NewImageFromBuffer(inputBuf)
+
+	if err != nil {
+		return []byte{}, fmt.Errorf("could not load image from url: %s", resp.Request.URL)
+	}
```

Comment:
Should use a logger derived with logger.FromContext() here. (Requires passing down the request context to BuildAnimatedThumbnail() but this is considered good practice anyway.)